src/GraphQL/ResolveFieldContext/IResolveFieldContext.cs

```diff
@@ -16,6 +16,15 @@ namespace GraphQL
     /// </summary>
     public interface IResolveFieldContext : IProvideUserContext
     {
+        /// <summary>
+        /// Gets an array of the specified minimum length from the pool.
+        /// This array does not need to be returned to the pool yourself.
+        /// </summary>
+        /// <typeparam name="TElement">Array element type.</typeparam>
+        /// <param name="minLength">The minimum length of the array.</param>
+        /// <returns>Array from pool.</returns>
+        TElement[] GetPooledArray<TElement>(int minLength);
```

Comment:
I have not found a better place for this method.

Reply:
I wish we could keep any memory pool usage `internal`.

Reply:
At minimum, I think we should document the lifetime explicitly.  e.g. "This array will be returned to the pool once the execution completes".  It is important that somebody does not try to use memory from this pool and store it in, for example, `Extensions`.

Reply:
> I wish we could keep any memory pool usage internal.

This method is deliberately placed in the public API, because it makes sense for consumers to use it for theis resolvers, and not just inside GraphQL.NET internal bits. I added more comments.